animesaturn.py

```diff
@@ -0,0 +1,336 @@
+        parsed_html = BeautifulSoup(pastebin_url,"html.parser")
+        self.titolo = url.split("/")[-1]
+        all_info = parsed_html.find('div', attrs={'class':'container shadow rounded bg-dark-as-box mb-3 p-3 w-100 text-white'})
+        info = re.findall("(?<=<b>Episodi:</b> )(.*)(?=<br/>)",str(all_info))
+        anime_type = parsed_html.find('span', attrs={'class':'badge badge-secondary'})
+        while (mutex):
+            time.sleep(0.5)
+        mutex = True
+        if ('OVA' in anime_type.text or "Special" in info[0]):
+            self.season_num = 0
+        elif "Movie" in info[0]:
+            self.season_num = -1
+        else:
+            self.season +=1
+            self.season_num = self.season
+        anime_ep = parsed_html.find_all('div', attrs={'class':'btn-group episodes-button episodi-link-button'})
+        self.list_link.clear()
+        for dim in anime_ep:
+            episode = dim.find('a')['href']
+            episode = episode +"§%d"%self.season_num
+            ep_list.append(episode)
+        mutex = False
+        with concurrent.futures.ThreadPoolExecutor(max_workers=len(ep_list)) as pool:
+            if(self.debug):
+                print(os.getpid())
+                for t in pool._threads:
+                    print(t)
+            pool.map(self.one_link, ep_list)
+        ep_list.clear()
+
+    def one_link(self, ep):
+        x = ep.split("§")
+        new_r = requests.get(url = x[0], params = {})
+        pastebin_url = new_r.text
+        parsed_html = BeautifulSoup(pastebin_url,"html.parser")
+        anime_page = parsed_html.find('div', attrs={'class':'card bg-dark-as-box-shadow text-white'})
+        is_link = anime_page.find('a')['href']
+        if 'watch' in is_link:
+            episode = is_link+'&s=alt'
+        self.all_ep[episode] = x[1]
+        self.list_link.append([episode,int(x[0].split("-")[-1])])
+
+
+    def sig_handler(self, _signo, _stack_frame):
+        print("\n")
+        kill_child_processes(os.getpid())
+        sys.exit(0)
+
+    def get_correlati(self, url):
+        is_lang = "-ITA" in url #controlla se il link supporta la lingua ita
+        #analizzo url e cerco la sezione "correlati" e richiamo la funzione per trovare gli episodi per gonuno di essi
+        new_r = requests.get(url = url, params = {})
+        pastebin_url = new_r.text
+        parsed_html = BeautifulSoup(pastebin_url,"html.parser")
+        correlati = parsed_html.find_all('div', attrs={'class':'owl-item anime-card-newanime main-anime-card'})
+        self.correlati_list.append(url)
+        for dim in correlati:
+            anime = dim.find('a')['href']
+            if(self.config["DEFAULT"].getboolean('only_ITA') and is_lang):
+                if("-ITA" in anime):
+                    self.correlati_list.append(anime)
+            else:
+                self.correlati_list.append(anime)
+        self.reorder_correlati()
+
+    def kill_child_processes(self,parent_pid, sig=signal.SIGTERM):
```

Comment:
```suggestion
    @staticmethod
    def kill_child_processes(parent_pid, sig=signal.SIGTERM):
```
Codacy: method could be a function